dashboard/modules/serve/serve_head.py

```diff
@@ -0,0 +1,101 @@
+import asyncio
+import logging
+from typing import Optional
+
+import aiohttp
+from aiohttp.web import Request, Response
+import ray
+
+import ray.dashboard.optional_utils as dashboard_optional_utils
+import ray.dashboard.utils as dashboard_utils
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+routes = dashboard_optional_utils.ClassMethodRouteTable
+
+
+class ServeHead(dashboard_utils.DashboardHeadModule):
+    def __init__(
+        self, dashboard_head, http_session: Optional[aiohttp.ClientSession] = None
+    ):
+        super().__init__(dashboard_head)
+        self._controller = None
+        self._controller_lock = asyncio.Lock()
+        self._http_session = http_session
+
+    @routes.get("/api/serve_head/applications/")
+    @dashboard_optional_utils.init_ray_and_catch_exceptions()
+    async def get_serve_instance_details(self, req: Request) -> Response:
```

Comment:
This code seems to be an exact copy of [the same method from `serve_agent.py`](https://github.com/ray-project/ray/blob/bf738b7e542d619ca798b1a4d44023ed8b77978c/dashboard/modules/serve/serve_agent.py#L79). Is it possible to pull this implementation into a separate helper function and call the same function here and in `serve_head.py`? That would prevent these two functions from going out of sync.

Reply:
> Nice work on these changes! I'm looking forward to using this. A few nits–
> 
> * Could we add AM/PM to the deployment times?
To be aligned with the rest of the dashboard, I prefer we keep 24 hour clock

> * Could we show the configs as YAML instead of JSON?
Yes

> * In the first image, we label the deployment timestamp as `Deployment time`. In the second image, we label the same timestamp as `Deployed at`. Could we use a consistent label? I have a slight preference for `Deployed at`.
The screenshot is out of date. We actually changed all the labels to "Last deployed at"

Reply:
@rkooo567 , actually I meant to start a discussion. How should we deal with this situation where we want to have an API on both the head process and the agent process?

1. Sharing the code implementation 
2. or head endpoint proxies to the agent on the head node?

Reply:
We synced offline. I think we should just do 2 because usually agent and dashboard have different environment (e.g., the agent calls ray.init, but the dashboard doesn't). If serve doesn't require ray.init, it is okay to just share the code but it might be cleaner to just always do 2.